mlflow/store/artifact/azure_data_lake_artifact_repo.py

```diff
@@ -129,3 +140,103 @@ def _download_file(self, remote_file_path, local_path):
+            self.log_artifact(src_file_path, artifact_subdir)
+
+    def _retryable_adls_function(self, func, artifact_file_path, **kwargs):
+        # Attempt to call the passed function.  Retry if the credentials have expired
+        try:
+            func(**kwargs)
+        except requests.HTTPError as e:
+            if e.response.status_code in [403]:
+                new_credentials = self._get_write_credential_infos([artifact_file_path])[0]
+                kwargs["sas_url"] = new_credentials.signed_uri
+                func(**kwargs)
+            else:
+                raise e
+
+    def _multipart_upload(self, credentials, src_file_path, artifact_file_path):
+        """
+        Uploads a file to a given Azure storage location using the ADLS gen2 API.
+        """
+        try:
+            headers = self._extract_headers_from_credentials(credentials.headers)
+            # try to create the file
+            self._retryable_adls_function(
+                func=put_adls_file_creation,
+                artifact_file_path=artifact_file_path,
+                sas_url=credentials.signed_uri,
+                headers=headers,
+            )
+            # next try to append the file
+            futures = {}
+            file_size = os.path.getsize(src_file_path)
+            num_chunks = _compute_num_chunks(src_file_path, _MULTIPART_UPLOAD_CHUNK_SIZE)
+            use_single_part_upload = num_chunks == 1
+            for index in range(num_chunks):
+                start_byte = index * _MULTIPART_UPLOAD_CHUNK_SIZE
+                future = self.chunk_thread_pool.submit(
+                    self._retryable_adls_function,
+                    func=patch_adls_file_upload,
+                    artifact_file_path=artifact_file_path,
+                    sas_url=credentials.signed_uri,
+                    local_file=src_file_path,
+                    start_byte=start_byte,
+                    size=_MULTIPART_UPLOAD_CHUNK_SIZE,
+                    position=start_byte,
+                    headers=headers,
+                    is_single=use_single_part_upload,
+                )
+                futures[future] = index
+
+            _, errors = _complete_futures(futures, src_file_path)
+            if errors:
+                raise MlflowException(
+                    f"Failed to upload at least one part of {artifact_file_path}. Errors: {errors}"
+                )
+
+            # finally try to flush the file
+            if not use_single_part_upload:
+                self._retryable_adls_function(
+                    func=patch_adls_flush,
+                    artifact_file_path=artifact_file_path,
+                    sas_url=credentials.signed_uri,
+                    position=file_size,
+                    headers=headers,
+                )
+        except Exception as err:
+            raise MlflowException(err)
+
+    def _get_presigned_uri(self, artifact_file_path):
```

Comment:
Given that this presigned URL is generated statically from the sas_token, I'm wondering if it still makes sense to have the same _retryable_adls_function setup as in the databricks_artifact_repo.py. Let me know what you think.